dartagnan/src/main/java/com/dat3m/dartagnan/program/Events.java

```diff
@@ -0,0 +1,355 @@
+    }
+
+    public static Local newLocal(Register register, ExprInterface expr) {
+        return newLocal(register, expr, -1);
+    }
+
+    public static CondJump newJump(BExpr cond, Label target) {
+        return new CondJump(cond, target);
+    }
+
+    public static CondJump newJumpUnless(ExprInterface cond, Label target) { return newJump(new BExprUn(BOpUn.NOT, cond), target); }
+
+    public static IfAsJump newIfJump(BExpr expr, Label label, Label end) {
+        return new IfAsJump(expr, label, end);
+    }
+
+    public static IfAsJump newIfJumpUnless(ExprInterface expr, Label label, Label end) {
+        return newIfJump(new BExprUn(BOpUn.NOT, expr), label, end);
+    }
+
+    public static CondJump newGoto(Label target) {
+        return newJump(BConst.TRUE, target);
+    }
+
+    public static CondJump newFakeCtrlDep(Register reg, Label target) { return newJump(new Atom(reg, COpBin.EQ, reg), target); }
+
+    public static Fence newFence(String name) {
+        return new Fence(name);
+    }
+
+    public static FenceOpt newFenceOpt(String name, String opt) {
+        return new FenceOpt(name, opt);
+    }
+
+    public static Init newInit(IExpr address, IConst value) {
+        return new Init(address, value);
+    }
+
+    public static Label newLabel(String name) {
+        return new Label(name);
+    }
+
+    public static Load newLoad(Register register, IExpr address, String mo, int cLine) {
+        return new Load(register, address, mo, cLine);
+    }
+
+    public static Load newLoad(Register register, IExpr address, String mo) {
+        return newLoad(register, address, mo, -1);
+    }
+
+    public static Store newStore(IExpr address, ExprInterface value, String mo, int cLine) {
+        return new Store(address, value, mo, cLine);
+    }
+
+    public static Store newStore(IExpr address, ExprInterface value, String mo) {
+        return newStore(address, value, mo, -1);
+    }
+
+    public static Cmp newCompare(IExpr left, IExpr right) {
+        return new Cmp(left, right);
+    }
+
+    public static Skip newSkip() {
+        return new Skip();
+    }
+
+    public static FunCall newFunctionCall(String funName, int cLine) {
+        return new FunCall(funName, cLine);
+    }
+
+    public static FunRet newFunctionReturn(String funName, int cLine) {
+        return new FunRet(funName, cLine);
+    }
+
+    public static RMWLoad newRMWLoad(Register reg, IExpr address, String mo) {
+        return new RMWLoad(reg, address, mo);
+    }
+
+    public static RMWStore newRMWStore(RMWLoad loadEvent, IExpr address, ExprInterface value, String mo) {
+        return new RMWStore(loadEvent, address, value, mo);
+    }
+
+    public static RMWLoadExclusive newRMWLoadExclusive(Register reg, IExpr address, String mo) {
+        return new RMWLoadExclusive(reg, address, mo);
+    }
+
+    public static RMWStoreExclusive newRMWStoreExclusive(IExpr address, ExprInterface value, String mo) {
```

Comment:
I think it would be better to keep a unique constructor for the `RMWStoreExclusive` event and make this method to call the one below with false as the last parameter.

Reply:
That was just a mistake. Will fix this.